dbt/adapters/databricks/connections.py

```diff
@@ -65,43 +70,45 @@ class DatabricksSQLConnectionWrapper(object):
     """Wrap a Databricks SQL connector in a way that no-ops transactions"""
     # https://forums.databricks.com/questions/2157/in-apache-spark-sql-can-we-roll-back-the-transacti.html  # noqa
 
-    def __init__(self, handle):
-        self.handle = handle
+    _conn: DatabricksSQLConnection
+    _cursor: Optional[DatabricksSQLCursor]
+
+    def __init__(self, conn: DatabricksSQLConnection):
+        self._conn = conn
         self._cursor = None
 
     def cursor(self):
-        self._cursor = self.handle.cursor()
+        self._cursor = self._conn.cursor()
         return self
 
     def cancel(self):
         if self._cursor:
-            # Handle bad response in the pyhive lib when
-            # the connection is cancelled
             try:
                 self._cursor.cancel()
-            except EnvironmentError as exc:
+            except OperationalError as exc:
                 logger.debug(
                     "Exception while cancelling query: {}".format(exc)
                 )
 
     def close(self):
         if self._cursor:
-            # Handle bad response in the pyhive lib when
-            # the connection is cancelled
             try:
                 self._cursor.close()
-            except EnvironmentError as exc:
+            except OperationalError as exc:
                 logger.debug(
                     "Exception while closing cursor: {}".format(exc)
                 )
+        self._conn.close()
```

Comment:
I'm wondering whether this is also needed in dbt-spark.

Reply:
@jtcohen6 Is it fine in dbt-spark? I suspect it has the same issue without any notice.

Reply:
Came here because I was seeing the same unclosed socket error as in #34. Tried the fix in this PR, worked like a charm! Then I saw I was tagged...

This isn't an issue we've noticed with the `dbt-spark` connector, but it's not so surprising to me, different cursors/connectors behave differently. Even with the change in #35, which will allow `dbt-databricks` to inherit (and avoid reimplementing) common functionality from `dbt-spark`, I would expect these methods to be points of departure

Reply:
Got it. Thanks for confirming!

Reply:
@jtcohen6 I can confirm I'm experiencing an issue with unclosed connections with dbt-spark and the fix is needed there as well

Reply:
@danielhaviv Thanks for opening the issue!